src/data-workspace/display.js

```diff
@@ -0,0 +1,123 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import { NoticeBox, CircularLoader } from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React, { useEffect } from 'react'
+import styles from './display.module.css'
+import { Table } from './table.js'
+
+const query = {
+    dataSetReport: {
+        resource: 'dataSetReport',
+        params: ({ ds, pe, ou }) => ({
+            ds,
+            pe,
+            ou,
+        }),
+    },
+}
+
+const Display = ({ workflowName, periodId, organisationUnitId, dataSetId }) => {
+    const { called, loading, data, error, refetch } = useDataQuery(query, {
+        lazy: true,
+    })
+    const tables = data?.dataSetReport
+    const fetchDataSet = () => {
+        refetch({
+            ds: dataSetId,
+            pe: periodId,
+            ou: organisationUnitId,
+        })
+    }
+
+    useEffect(() => {
+        if (!dataSetId) {
+            return
+        }
+        fetchDataSet()
+    }, [dataSetId])
```

Comment:
This is invalid, we can only return a function ("on unmount" callback) in `useEffect`s

```suggestion
    useEffect(() => {
        if (!dataSetId) {
            fetchDataSet()
        }
    }, [dataSetId])
```